packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -0,0 +1,289 @@
+import * as React from 'react';
+import { composeEventHandlers } from '@radix-ui/primitive';
+import { useComposedRefs } from '@radix-ui/react-compose-refs';
+import { createContext } from '@radix-ui/react-context';
+import { useControllableState } from '@radix-ui/react-use-controllable-state';
+import * as PopperPrimitive from '@radix-ui/react-popper';
+import { Portal } from '@radix-ui/react-portal';
+import { Presence } from '@radix-ui/react-presence';
+import { Primitive, extendPrimitive } from '@radix-ui/react-primitive';
+import { useId } from '@radix-ui/react-id';
+import { Slot } from '@radix-ui/react-slot';
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+
+/* -------------------------------------------------------------------------------------------------
+ * HoverCard
+ * -----------------------------------------------------------------------------------------------*/
+
+const HOVERCARD_NAME = 'HoverCard';
+
+type HoverCardContextValue = {
+  triggerRef: React.RefObject<HTMLAnchorElement>;
+  contentId: string;
+  open: boolean;
+  onOpenChange(open: boolean): void;
+  onMouseEnter(): void;
+  onMouseLeave(): void;
+  hasCustomAnchor: boolean;
+  onCustomAnchorAdd(): void;
+  onCustomAnchorRemove(): void;
+};
+
+const [HoverCardProvider, useHoverCardContext] = createContext<HoverCardContextValue>(
+  HOVERCARD_NAME
+);
+
+type HoverCardOwnProps = {
+  open?: boolean;
+  defaultOpen?: boolean;
+  enterDelayDuration?: number;
+  exitDelayDuration?: number;
+  onOpenChange?: (open: boolean) => void;
+};
+
+const HoverCard: React.FC<HoverCardOwnProps> = (props) => {
+  const {
+    children,
+    open: openProp,
+    defaultOpen,
+    onOpenChange,
+    enterDelayDuration = 700,
+    exitDelayDuration = 400,
+  } = props;
+  const triggerRef = React.useRef<HTMLAnchorElement>(null);
+  const [hasCustomAnchor, setHasCustomAnchor] = React.useState(false);
```

Comment:
I'm not sure if a `HoverCard` should have a custom anchor given that we're saying it's specifically to preview content behind a link. Would consumers ever need a link preview to be anchored to something else?

cc: @benoitgrelard

Reply:
My thought is that it’s hard to predict the layout challenges some users may have, and that a custom anchor does provide an escape hatch if they have some strange arrangement where the positioning props alone wouldn’t solve it. 🤔 

Depends how opinionated we want to be here.

Reply:
We tend to prefer simplicity on our side for first releases and then add functionality as/when consumers request it if their use-case cannot be achieved another way since adding stuff prematurely and removing it later is usually harder.

Reply:
I have been wondering about exactly this too…
I feel like we don't have a clear formed idea yet as to when to provide this functionality vs not.
But given that we've removed it from `DropdownMenu` and `Tooltip` and I feel like this is closer to `Tooltip` than it is from `Popover` perhaps we should not add it.
It's also our opportunity to start with less and only add if someone asks for it, this way we can see what the use cases are.

Reply:
Yep +1. My preference is always to avoid predicting what problems users may have as it can often cause unnecessary bloat in a codebase, and instead react to actual problems they report.

Reply:
But then we also don't know if people are using it in `Popover` so… 😀

Reply:
> It's also our opportunity to start with less and only add if someone asks for it, this way we can see what the use cases are.

> My preference is always to avoid predicting what problems users may have as it can often cause unnecessary bloat in a codebase, and instead react to actual problems they report.

I think this is a great approach, I am convinced 😁

Reply:
> But then we also don't know if people are using it in `Popover` so…

@benoitgrelard Oh really? I thought we had it there because composer needed it 😬  does it not?